src/app.rs

```diff
@@ -106,13 +106,17 @@ mod tests {
     }
 }
 
+/* Claims are used in two forms, one for API calls, and one for
+ * general repo access, the second one is simpler and just uses scope
+ * for the allowed ids, and sub means the user doing the access (which
+ * is not verified). */
 #[derive(Clone, Debug, Serialize, Deserialize)]
 pub struct Claims {
-    pub sub: String, // "build", "build/N"
-    pub scope: Vec<String>, // "build", "upload" "publish"
+    pub sub: String, // "build", "build/N", or user id for repo tokens
+    pub scope: Vec<String>, // "build", "upload" "publish", or list of id prefixes for repo tokens
```

Comment:
This is a bit pedantic, but `scope` is actually a registered claim for OAuth tokens (https://www.iana.org/assignments/jwt/jwt.xhtml). It's not a requirement for JWT and could be made optional with the "private" `prefixes` claim used consistently between build and repo tokens.

Reply:
I don't understand the issue here. I used scope precisely because that is typically what is used for these kinds of things. Its registered by iana to mean this: https://tools.ietf.org/html/rfc6749#section-3.3 and to my eyes it looks like we're following that.

Reply:
Oh, you mean we could use prefix for both and no scope for the repo tokens? I guess that might be better.

Reply:
What I'm saying is that `scope` is a claim registered for OAuth tokens (as you linked to), but this isn't an OAuth token. It probably doesn't matter and may be more pain than it's worth, but I think it would be more appropriate to use a custom claim name like `flatpak_usage` or `xa_usage` or something like that.

Reading it all again, I guess there's no issue with using the OAuth registered `scope` claim as long as the specified format is used. The specified value format is pretty flexible - a space separated list of scope tokens where the scope token is any printable ASCII character except `"` and `\`. So, I drop my objection here.